### PR TITLE
[BigQuery] Extend timestamp precision to microseconds when writing with Beam Rows (Storage API)

### LINKED ISSUE #24782: [Bug]: Timestamp microseconds are cut off when writing Beam Rows to BigQuery via Storage API.
### What happened?

When writing Beam Rows to BigQuery via Storage API, my Timestamp elements are arriving to the table with only second precision. Note: this specifically happens when using Beam schema (setting a TableSchema via `.withSchema()` works fine).

The pipeline below reproduces this:
```
public static void main(String[] args) {
  List<Instant> times = Arrays.asList(
      Instant.ofEpochSecond(100, 10 * 1000),
      Instant.ofEpochSecond(200, 20 * 1000),
      Instant.ofEpochSecond(300, 30 * 1000));

  Schema schema = Schema.of(Field.of("time", FieldType.logicalType(SqlTypes.TIMESTAMP)));
  Pipeline p = Pipeline.create();
  String table = "<project>:<dataset>.<table>";
  p
      .apply(Create.of(times))
      .apply(MapElements.into(TypeDescriptor.of(Row.class)).via(
          time -> Row.withSchema(schema).withFieldValue("time", time).build()
      )).setRowSchema(schema)
      .apply(BigQueryIO.<Row>write().to(table)
          .useBeamSchema()
          .withMethod(Write.Method.STORAGE_WRITE_API)
          .withFormatFunction(BigQueryUtils.toTableRow()));

  p.run().waitUntilFinish();
}
```

I expect the following values to be written:
```
1970-01-01T00:05:00.000030
1970-01-01T00:03:20.000020 
1970-01-01T00:01:40.000010
```

But instead I get these:
```
1970-01-01 00:05:00 
1970-01-01 00:03:20  
1970-01-01 00:01:40 
```

### Issue Priority

Priority: 1 (data loss / total loss of function)

### Issue Components

- [ ] Component: Python SDK
- [X] Component: Java SDK
- [ ] Component: Go SDK
- [ ] Component: Typescript SDK
- [X] Component: IO connector
- [ ] Component: Beam examples
- [ ] Component: Beam playground
- [ ] Component: Beam katas
- [ ] Component: Website
- [ ] Component: Spark Runner
- [ ] Component: Flink Runner
- [ ] Component: Samza Runner
- [ ] Component: Twister2 Runner
- [ ] Component: Hazelcast Jet Runner
- [ ] Component: Google Cloud Dataflow Runner

### PR DESCRIPTION
**Pull request by ahmedabu98**
Fixes #24782

Timestamp logical type in Beam Row is being encoded in millisecond precision, see [here](https://github.com/apache/beam/blob/820839fe86b16c4600b0aab4ad8236456cd200a9/sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BeamRowToStorageApiProto.java#L122). These changes update it to encode in microsecond precision.

### CODE REVIEW

**Review by Abacn — COMMENTED** [review]
We may also adjust the test case here to feed a micro precision timestamp. https://github.com/apache/beam/blob/820839fe86b16c4600b0aab4ad8236456cd200a9/sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/bigquery/BeamRowToStorageApiProtoTest.java#L267

**Comment by ahmedabu98** [comment]
Ahh looks like `Instant.now()` [1] is using system UTC clock [2], which uses millisecond precision. Will add a test here for microsecond precision, thanks for pointing it out @Abacn.

[1] https://docs.oracle.com/javase/8/docs/api/java/time/Instant.html#now--
[2] https://docs.oracle.com/javase/8/docs/api/java/time/Clock.html#systemUTC--

File: sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/bigquery/BeamRowToStorageApiProtoTest.java [hunk 1]
```diff
@@ -392,4 +415,11 @@ public void testMessageFromTableRow() throws Exception {
     DynamicMessage nestedMsg = (DynamicMessage) msg.getField(fieldDescriptors.get("nested"));
     assertBaseRecord(nestedMsg);
   }
+
+  @Test
```
Comment (Abacn) [inline sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/bigquery/BeamRowToStorageApiProtoTest.java:419]: Thanks for the finding. However this test looks irrelevant to the class we test against. We could simply replace "sqlTimestampValue" with a constant timestamp with micros precision, like those "int64Value", "byteValue" in BASE_ROW

Comment (ahmedabu98) [reply sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/bigquery/BeamRowToStorageApiProtoTest.java:419]: Ahh yes, I wrote that test just to confirm that the system UTC clock on the Jenkins machine was in fact using millis precision.

I logged the output of `Instant.now()` (see [here](https://ci-beam.apache.org/job/beam_PreCommit_Java_GCP_IO_Direct_Commit/1193/testReport/org.apache.beam.sdk.io.gcp.bigquery/BeamRowToStorageApiProtoTest/testInstant/)) and got `2022-12-27T19:32:26.036Z`

Comment (ahmedabu98) [reply sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/bigquery/BeamRowToStorageApiProtoTest.java:419]: will remove the test now!

Comment (ahmedabu98) [reply sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/bigquery/BeamRowToStorageApiProtoTest.java:419]: +1 to extending sqlTimestampValue micros precision

**Review by Abacn — APPROVED** [review]
Run Java_GCP_IO_Direct PreCommit

**Comment by Abacn** [comment]
Run Java_GCP_IO_Direct PreCommit
